### PR TITLE
PostgreSQL: add initial support (#4241) - v9

### PR DESCRIPTION
**Pull request by jufajardini**
Link to redmine ticket: https://redmine.openinfosecfoundation.org/issues/4241

Previous PR: #6799

Describe changes from previous PR:
- fix scan-build warnings about uninitialized values in output-json-pgsql. 

The idea is to squash that up, left it separately to make it easier to review - in case that's not a valid fix. (although it worked locally).

suricata-verify-pr: 656
OISF/suricata-verify#656

### CODE REVIEW

File: rust/src/pgsql/pgsql.rs [hunk 1]
```diff
@@ -0,0 +1,879 @@
...
+    /// If there is data from the backend message that Suri should store separately in the State or
+    /// Transaction, that is also done here
+    fn response_process_next_state(
+        &mut self, response: &PgsqlBEMessage, f: *const Flow,
+    ) -> Option<PgsqlStateProgress> {
+        match response {
+            PgsqlBEMessage::SSLResponse(parser::SSLResponseMessage::SSLAccepted) => {
+                SCLogDebug!("SSL Request accepted");
+                unsafe {
+                    AppLayerRequestProtocolTLSUpgrade(f);
```
Comment (jufajardini) [inline rust/src/pgsql/pgsql.rs:348]: Should this be changed to use our rust TLS parser?

**Review by jufajardini — COMMENTED** [review]
Apparently rebasing with the newest master this leads to conflicts, gonna check this tomorrow...

**Comment by suricata-qa** [comment]
ERROR: QA failed on build_asan.
  

Pipeline 5780

File: rust/src/pgsql/pgsql.rs (hunk 1 above)
Comment (victorjulien) [reply rust/src/pgsql/pgsql.rs:348]: No all this does it tell the app layer api to see if the flow can switch to tls, so no need to worry about handling the tls itself.

Comment (jufajardini) [reply rust/src/pgsql/pgsql.rs:348]: oki, thanks!

**Comment by jufajardini** [comment]
replaced by: https://github.com/OISF/suricata/pull/6817

**Pull request closed without merging**